java-frontend/src/main/java/org/sonar/java/se/constraint/BooleanConstraint.java

```diff
@@ -46,6 +50,15 @@ public String valueAsString() {
     return "false";
   }
 
+  @Nullable
+  @Override
+  public Constraint copyOver(RelationalSymbolicValue.Kind kind) {
+    if(kind == GREATER_THAN_OR_EQUAL || kind == LESS_THAN) {
+      return null;
+    }
+    return Constraint.super.copyOver(kind);
```

Comment:
I think it will be more readable to inline logic from super here
```
 switch (kind) {
      case EQUAL:
      case METHOD_EQUALS:
        return this;
      case LESS_THAN:
      case GREATER_THAN_OR_EQUAL:
        return null;
      default:
        return inverse();
    }
```